### DISCUSSION TITLE
Supabase User Sessions not restricting multiple logins

### DISCUSSION THREAD

**Discussion by Matignaciom** [supabase/supabase#26358] [Questions] [answered]
**Question**:
Hello! I recently activated the User Sessions functionality in Supabase, which should limit my Flutter app to only one logged-in user at a time (one session per user). However, it doesn't seem to be working. When I try to log in to my app from another device using the same Supabase authentication, the previous session isn't closed. What could be the issue?

**Example**:
I've implemented Supabase authentication in my Flutter app, and I want to restrict users to only one active session at a time. Even though I've enabled the User Sessions functionality, multiple logins are still allowed. For instance, when I log in to my app from my phone, I can still log in from my tablet using the same credentials without the first session being closed automatically. What am I missing in the setup?

![image](https://github.com/supabase/supabase/assets/66745824/7e5e3749-77ce-4e26-b189-e1eec791504e)

**Comment by GaryAustin1** [comment, marked as answer]
Sessions can only be checked when the jwt expires.  There is no ability to "signout" a device until it asks for a new session by doing refresh or login.  So your first login client will remain active it until the jwt expires.  Then it will have its session destroyed.

![image](https://github.com/supabase/supabase/assets/54564956/5147b1fe-9072-4649-8cc0-1cf4f0e22ffa)
https://supabase.com/docs/guides/auth/sessions

**Reply by Matignaciom** [reply]
Thank you very much for your response, but I have a question. How could I manage this to ensure that there are no two active login sessions on two different devices? I've been looking at the API in "User Management" for GET USER and LOG OUT, but I'm not sure how to handle it. Could you please guide me if it's not too much trouble? Thanks in advance.

**Reply by GaryAustin1** [reply]
There is no way to notify the client their session has changed status in a REST API.
You would need to use RLS somehow, poll from your client, make your jwt shorter (5minutes) or find some other means to pull this off.
Nothing built in to immediately close a client.

**Reply by Matignaciom** [reply]
Thank you very much for your response. I will have to implement some way from login and logout with a field in the Supabase database. I am already testing it, and I think I have it solved. I have marked your solution. Thank you very much! :D
